src/main/cpp/kernels/SystemML.cu

```diff
@@ -28,6 +28,27 @@ nvcc -ptx -arch=sm_30 SystemML.cu
 
 
 /**
+ * Performs a slice operation where the input matrix is sparse and the output matrix is dense.
+ * This function avoids unnecessary sparse to dense conversion of the input matrix.
+ */
+extern "C"
```

Comment:
A little javadoc explaining the overall logic and the parameters would be nice here.